plat/linuxu/memory.c

```diff
@@ -127,10 +127,13 @@ static int __linuxu_plat_initrd_init(void)
 
 int ukplat_memregion_count(void)
```

Comment:
Oh and one more thing. Based on your commit description, the change to `ukplat_memregion_count` should be in a separate commit, as it does not fix anything, but rather just adds a minor improvement. So, I would suggest you place this in a commit with a message like `Avoid re-init of heap and initrd in ukplat_memregion_count` or something like that and the actual fix with your current message in a separate commit. This way, **this** PR should have 2 commits instead of one: a commit with your current message and the changes done to `ukplat_memregion_get` (your actual fix) and a commit with your changes to `ukplat_memregion_count` (the improvement). This is how I would see it, just my opinion ;).

Reply:
I agree with @mogasergiu